samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -103,23 +103,58 @@ def extract_cors(self, cors_prop):
         """
         cors = None
         if cors_prop and isinstance(cors_prop, dict):
-            allow_methods = cors_prop.get("AllowMethods", ",".join(sorted(Route.ANY_HTTP_METHODS)))
-            allow_methods = self.normalize_cors_allow_methods(allow_methods)
+            allow_methods = self._get_cors_prop(cors_prop, "AllowMethods")
+            if allow_methods:
+                allow_methods = self.normalize_cors_allow_methods(allow_methods)
+            else:
+                allow_methods = ",".join(sorted(Route.ANY_HTTP_METHODS))
+
+            allow_origin = self._get_cors_prop(cors_prop, "AllowOrigin")
+            allow_headers = self._get_cors_prop(cors_prop, "AllowHeaders")
+            max_age = self._get_cors_prop(cors_prop, "MaxAge")
+
             cors = Cors(
-                allow_origin=cors_prop.get("AllowOrigin"),
-                allow_methods=allow_methods,
-                allow_headers=cors_prop.get("AllowHeaders"),
-                max_age=cors_prop.get("MaxAge"),
+                allow_origin=allow_origin, allow_methods=allow_methods, allow_headers=allow_headers, max_age=max_age
             )
         elif cors_prop and isinstance(cors_prop, string_types):
+            allow_origin = cors_prop
+            if allow_origin[0] != "'" or allow_origin[-1] != "'":
```

Comment:
if this is a string, can we just do startswith or endswith instead?

Reply:
Will fix!